ui/dashboards/src/components/PanelDrawer/PanelDrawer.tsx

```diff
@@ -41,33 +41,44 @@ const PanelDrawer = () => {
   const { panels, updatePanel } = usePanels();
   const { panelDrawer, closePanelDrawer } = useDashboardApp();
 
+  const panelKey = panelDrawer?.panelKey;
+  const panelDefinition = panelKey ? panels[panelKey] : undefined;
+
   let defaultPanelName = '';
-  let defaultDescription = '';
-  if (panelDrawer?.panelKey) {
+  let defaultPanelDescription = '';
+  let defaultKind = '';
+  let defaultOptions;
+  if (panelDefinition) {
     // editing an existing panel
-    defaultPanelName = panels[panelDrawer.panelKey]?.spec.display.name ?? '';
-    defaultDescription = panels[panelDrawer.panelKey]?.spec.display.description ?? '';
+    defaultPanelName = panelDefinition.spec.display.name ?? '';
+    defaultPanelDescription = panelDefinition.spec.display.description ?? '';
+    defaultKind = panelDefinition.spec.plugin.kind ?? '';
+    defaultOptions = panelDefinition.spec.plugin.spec;
   }
   const [group, setGroup] = useState(panelDrawer?.groupIndex);
   const [panelName, setPanelName] = useState(defaultPanelName);
-  const [panelDescription, setPanelDescription] = useState(defaultDescription);
-  const [kind, setKind] = useState('');
-  const [options, setOptions] = useState<unknown>({});
+  const [panelDescription, setPanelDescription] = useState(defaultPanelDescription);
+  const [kind, setKind] = useState(defaultKind);
+  const [options, setOptions] = useState<unknown>(defaultOptions);
 
   // TO DO: we might want to make the form a sub component we don't need this useEffect
   // currently, we need to reset the states whenever panelDrawer is reopened
   // since this component does not get remounted when it open/closes (otherwise we lose the animation of sliding in/out)
   useEffect(() => {
```

Comment:
Might look into following this comment so we don't need this useEffect, it feels strange to set the values twice.

Reply:
Yeah, agreed. This was something that Eunice and I talked about during code review, so she left a TODO. If you want to try and tackle it, go for it.